File: z-clients/python/diom/models/sasl_mechanism.py
# this file is @generated
from enum import Enum


class SaslMechanism(str, Enum):
    """The SASL mechanism, mapped onto librdkafka's `sasl.mechanism`."""

    PLAIN = "plain"
    SCRAM_SHA256 = "scram-sha256"
    SCRAM_SHA512 = "scram-sha512"

    def __str__(self) -> str:
        return str(self.value)

File: z-clients/java/src/main/java/com/svix/diom/models/SaslMechanism.java
// this file is @generated
package com.svix.diom.models;

import com.fasterxml.jackson.annotation.JsonValue;

public enum SaslMechanism {
    PLAIN("plain"),
    SCRAM_SHA256("scram-sha256"),
    SCRAM_SHA512("scram-sha512");
    private final String value;

    SaslMechanism(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }
}